manifests/agent/cron.pp

```diff
@@ -16,9 +17,13 @@
     $cmd = "${puppet::params::puppet_cmd} agent --confdir ${puppet::params::puppet_confdir} --onetime --no-daemonize >/dev/null"
   }
 
+  $interval = 60 / $frequency
+  $random_offset = fqdn_rand($interval)
+  $cron_schedule = $frequency.map |$value| { ($value * $interval) + $random_offset}
```

Comment:
gosh, i really want to switch.
i wonder how many of the modules i use would cough up blood the moment i do…